src/components/Table/TableHeaderComponent.jsx

```diff
@@ -73,7 +102,9 @@ TableHeader.contextTypes = {
 
 TableHeader.propTypes = {
   visibleColumns: PropTypes.array.isRequired,
-  properties: PropTypes.object.isRequired
+  properties: PropTypes.object.isRequired,
+  checkedAll: PropTypes.object.isRequired,
+  checkAll: PropTypes.func.isRequired
```

Comment:
Why isn't checkedAll just a flag. Instead it's implemented as an Object which is just another thing everyone making changes to it will have to keep in mind.